### ISSUE TITLE
Suggestion: popup question if player wants to recall spy to draw a card.

### ISSUE THREAD

**Issue by supertopi** [rogeriocamorim/duneBugSuggestionTracker#9]
In Uprising when sending agent to a board space, recalling spy to draw a card should be done before any other actions. 
Currently with the click space automation, this can feel clunky.

Suggestion to add a popup question if player wants to recall spy to draw a card, when sending agent to a board space that they are spying on. This would only be necessary when sending agent to an unoccupied board space.

**Comment by Chatanga** [comment]
The solution currently in testing is to have the right click on a space doing both actions (in case a spy is available). The recalled spy is arbitrary in the case of the Research Station, but it could easily be fixed manually if it is not the right one.

**Comment by dotdotdot-tts** [comment]
while this is a clean solution, it just feels less intuitive. a pop-up to ask player whether they want to draw a card for the spy before doing the space actions would be much more intuitive.

it's a very similar reasoning for the pop-up for when the mentat is missing to warn the player.

**Comment by Chatanga** [comment]
Since willingly sending an agent to get a missing Mentat is rare, a popup is appropriate in this case. Sending an agent to a space with a spy but no intention of using it is a lot more common and a popup could be annoying. After all, it's the reason why you ask to remove it for the Spice Refinery.

**Comment by dotdotdot-tts** [comment]
i agree it can be annoying, however the spy thing is a much more impactful thing in terms of gameplay since if they draw from a space like fremkit or espionage, they miss their opportunity to draw from the spy.

i think in the case of the spy, it's warranted to minimize gameplay issues.

**Comment by dotdotdot-tts** [comment]
> The solution currently in testing is to have the right click on a space doing both actions (in case a spy is available). The recalled spy is arbitrary in the case of the Research Station, but it could easily be fixed manually if it is not the right one.

i just want to add that you cannot manually fix by drawing another card for the spy if there was no declaration that you are removing the spy. basically allows a player to see 2 cards from the draw and then go "oh i forgot spy" and draw another, which is against the rules.

**Comment by Chatanga** [comment]
I forgot the case where you want to recall 2 spies to the Research Station... The manual solution that I mentioned was simply to exchange the 2 spies if it was not the correct one that was recalled. And on the other hand, it's not perfect either, since the card drawn could help you choose which spy you want to recall. I guess a popup is needed for this space in case 2 spies are available. As for adding one for all spaces offering a card draw, I'm still not a big fan, but it could be tested.

**Comment by aaittouares** [comment]
Dialog are annoying for a lot of "hardcore" Dune players and i'm not keen to have it since i predict a complaint if we put a dialog there.
Right click is maybe not obvious but once you know it, it's easy and not invasive.
We'll put those shortcuts in the description of the mod.

**Comment by dotdotdot-tts** [comment]
i am one of those hardcore dune players and i haven't heard anyone complain about a possible pop-up for the spy draw when discussed. 

in fact, it's viewed very favorably since if you reject the dialog box, it is definitive that you did not want to do draw for the spy and cannot say something like "i meant to draw a card for the spy" after seeing what cards you drew. this is important for the integrity of the game, which i stress is more important than anything. it actually is a critical decision that has gameplay consequences.

also the right-click method, introduces the "i meant to right click to draw for the spy" excuse as well. it is not a cleaner gameplay solution.

in most cases, i will accept devs' choice, however this one is too important to gameplay, i respectfully insist you reconsider.

**Comment by aaittouares** [comment]
i didn't say i would not consider it
my concern is really about the community response to such a decision
you are not alone on this so what i propose is we make some discussion on the discord about your suggestion and see what happens

**Comment by dotdotdot-tts** [comment]
ah fair, i misunderstood what you meant by that you're not keen to have it haha.

sounds good! discussion is good

**Comment by dotdotdot-tts** [comment]
since the pop-up annoyance is a concern, we don't really need 1 for every spot; i think it's only necessary for spots that unveil new information: card draw, intrigue, contract

**Comment by Chatanga** [comment]
Fixed in [Fixed issues:](https://github.com/aaittouares/Dune_Uprising_TTS/commit/e68cdf5f44e2b1950e3d8d695fadf463d477e490)

**Comment by supertopi** [comment]
> Fixed in [Fixed issues:](https://github.com/aaittouares/Dune_Uprising_TTS/commit/e68cdf5f44e2b1950e3d8d695fadf463d477e490)

broken link, or perhaps no access to project. Anyway thanks for your support!

**Comment by aaittouares** [comment]
> > Fixed in [Fixed issues:](https://github.com/aaittouares/Dune_Uprising_TTS/commit/e68cdf5f44e2b1950e3d8d695fadf463d477e490)
> 
> broken link, or perhaps no access to project. Anyway thanks for your support!

you have indeed no access to the project. it's a link meant only for the mod's team :)

**Comment by aaittouares** [comment]
changed to https://github.com/aaittouares/Dune_Uprising_TTS/commit/61e4942a1f5ca4690ffa99df3c5ecb4a2f17a9b6 to better reflect the use case described by players

**Comment by dotdotdot-tts** [comment]
amazing work!
